integration-tests/api/__tests__/admin/price-list.js

```diff
@@ -0,0 +1,712 @@
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      ])
+    })
+
+    it("Adds a batch of new prices to a price list overriding existing prices", async () => {
+      const api = useApi()
+
+      const payload = {
+        prices: [
+          {
+            amount: 45,
+            currency_code: "usd",
+            variant_id: "test-variant",
+            min_quantity: 1001,
+            max_quantity: 2000,
+          },
+          {
+            amount: 35,
+            currency_code: "usd",
+            variant_id: "test-variant",
+            min_quantity: 2001,
+            max_quantity: 3000,
+          },
+          {
+            amount: 25,
+            currency_code: "usd",
+            variant_id: "test-variant",
+            min_quantity: 3001,
+            max_quantity: 4000,
+          },
+        ],
+        override: true,
+      }
+
+      const response = await api
+        .post(
+          "/admin/price-lists/pl_no_customer_groups/prices/batch",
+          payload,
+          {
+            headers: {
+              Authorization: "Bearer test_token",
+            },
+          }
+        )
+        .catch((err) => {
+          console.warn(err.response.data)
+        })
+
+      expect(response.status).toEqual(200)
+      expect(response.data.price_list.prices.length).toEqual(3)
+      expect(response.data.price_list.prices).toMatchSnapshot([
+        {
+          id: expect.any(String),
+          price_list_id: "pl_no_customer_groups",
+          amount: 45,
+          currency_code: "usd",
+          variant_id: "test-variant",
+          min_quantity: 1001,
+          max_quantity: 2000,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+        {
+          id: expect.any(String),
+          price_list_id: "pl_no_customer_groups",
+          amount: 35,
+          currency_code: "usd",
+          variant_id: "test-variant",
+          min_quantity: 2001,
+          max_quantity: 3000,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+        {
+          id: expect.any(String),
+          price_list_id: "pl_no_customer_groups",
+          amount: 25,
+          currency_code: "usd",
+          variant_id: "test-variant",
+          min_quantity: 3001,
+          max_quantity: 4000,
+          created_at: expect.any(String),
+          updated_at: expect.any(String),
+        },
+      ])
+    })
+  })
+
+  describe("DELETE /admin/price-lists/:id", () => {
+    beforeEach(async () => {
+      try {
+        await adminSeeder(dbConnection)
+        await productSeeder(dbConnection)
+        await priceListSeeder(dbConnection)
+      } catch (err) {
+        console.log(err)
+        throw err
+      }
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("Deletes a price list", async () => {
```

Comment:
Can we add another more test cases for deletion with price lists to ensure we handle delete behaviour correctly wrt. variants and money amounts, specifically that when we delete a variant the money amounts are also removed from the pricelist

Reply:
Good catch @pKorsholm ! We don't appear to handle this atm, if you delete a Product Variant it does not affect it's prices they will continue to exist, and as such also stay part of any PriceList they are associated with. Perhaps we should ensure that `money_amout where variant_id = '<deleted_variant_id>'` are deleted after a Variant is deleted

Reply:
making the following change to `ProductVariantService`, fixes this:

```ts
/**
   * Deletes variant.
   * Will never fail due to delete being idempotent.
   * @param {string} variantId - the id of the variant to delete. Must be
   *   castable as an ObjectId
   * @return {Promise<void>} empty promise
   */
  async delete(variantId: string): Promise<void> {
    return this.atomicPhase_(async (manager: EntityManager) => {
      const variantRepo = manager.getCustomRepository(
        this.productVariantRepository_
      )

      const variant = await variantRepo.findOne({
        where: { id: variantId },
        relations: ["prices"], // by loading the prices we ensure that the softRemove is cascaded
      })

      if (!variant) {
        return Promise.resolve()
      }

      await variantRepo.softRemove(variant)

      await this.eventBus_
        .withTransaction(manager)
        .emit(ProductVariantService.Events.DELETED, {
          id: variant.id,
          product_id: variant.product_id,
          metadata: variant.metadata,
        })

      return Promise.resolve()
    })
  }
```